client/src/ui/utils/Camera.tsx

```diff
@@ -105,8 +109,8 @@ const CameraControls = ({ position, target }: Props) => {
       _v.sub(e?.target.target);
       camera.position.sub(_v);
 
-      // setCameraPosition(camera.position.clone());
-      // setCameraTarget(ref.current.target.clone());
+      setCameraPosition({ ...camera.position.clone(), muteSound: true });
+      setCameraTarget({ ...ref.current.target.clone(), muteSound: true });
```

Comment:
I'm not sure if these two lines should be here. Let's remove them.

**cameraPosition** and **cameraTarget** in the zustand store are used to trigger automatic transition from current camera position to the new one E.g. wen user clicks "show on map" - he will see an animated transition to that entity. So **cameraPosition** and **cameraTarget** do not always storing current actual camera values, but they are used only once at the moment of transition (and usually they are not changed directly, but by calling a function like moveCameraToRealmId)

And clampPan function is just restricts user to move outside of the map bounds, there's no need to set camera position and target every time the user pans. Also I see one improvement that we can make - let's add a condition that the clampPan is executed only if the user is really trying to go off the map bounds, instead of clamping every time user pans, even if the camera does not go off the bounds.

Reply:
You guys know best but

if we want to have a compass then we need to store the current camera position and target somewhere. The clampPan function triggers when the user rotate/move the camera. Problem was "flySound" got triggered with change to cameraPosition.

If you remove those lines I would suggest also removing the compass

Reply:
> You guys know best but
> 
> if we want to have a compass then we need to store the current camera position and target somewhere. The clampPan function triggers when the user rotate/move the camera. Problem was "flySound" got triggered with change to cameraPosition.
> 
> If you remove those lines I would suggest also removing the compass

Got your point. I moved code for calculating direction of compass to MapControls onChange handler and added throttling for it. Now it should work fine. There were two reasons why I didn't want to set camera pos & target - 1) this a trigger for statring camera animation 2) onChange is called very frequently and could lead to performance drops if combined with reassigning variables and recalculating direction value

Reply:
Nice, looks good!